core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -897,6 +897,80 @@ class SparkSubmitSuite
     sysProps("spark.submit.pyFiles") should (startWith("/"))
   }
 
+  test("handle remote http(s) resources in yarn mode") {
```

Comment:
It seems you have 3 different tests in this block (at least), could you break them into separate tests?